ledger/complete/ledger.go

```diff
@@ -368,3 +376,26 @@ func (l *Ledger) DumpTrieAsJSON(state ledger.State, outputFilePath string) error
 
 	return trie.DumpAsJSON(writer)
 }
+
+// this operation should only be used for exporting
+func (l *Ledger) keepOnlyOneTrie(state ledger.State) error {
+	// don't write things to WALs
+	l.wal.PauseRecord()
+	defer l.wal.UnpauseRecord()
+
+	allTries, err := l.forest.GetTries()
+	if err != nil {
+		return err
+	}
+
+	targetRootHash := ledger.RootHash(state)
+	for _, trie := range allTries {
+		trieRootHash := trie.RootHash()
+		if !bytes.Equal(trieRootHash, targetRootHash) {
+			l.forest.RemoveTrie(trieRootHash)
+		}
+	}
+	// call gc to free up mem
+	runtime.GC()
```

Comment:
I think we are trying to call the GC less often, as @zhangchiqing was researching. Don't you think it's fine to keep the heap full and then free it at the next automatic GC call ?

Reply:
I think it's not necessary. 

Yeah, we have triggered GC too often. We need to reduce rather than increase the time of calling GC.

And as you mentioned, this is in the spork prepare step, then no need to free memory, as the problem will release all after it exits